### PR TITLE
Updated marathon and bamboo configs on Dev to use https endpoint

### PR DESCRIPTION
**Pull request by amitbose327**

### CODE REVIEW

File: ansible/roles/marathon-server/templates/marathon-flags/webui_url.j2 [hunk 1]
```diff
@@ -1 +1 @@
-http://{{ marathon_webui_host }}:{{ marathon_webui_port }}/
+{% if bamboo_marathon_proxy_scheme is defined %}{{ bamboo_marathon_proxy_scheme }}{% else %}http{% endif %}://{{ marathon_webui_host }}:{{ marathon_webui_port }}/
```
Comment (ankanm) [inline ansible/roles/marathon-server/templates/marathon-flags/webui_url.j2]: Consider replacing if condition with - `{{ bamboo_marathon_proxy_scheme | default('http') }}`

File: ansible/roles/bamboo/templates/production.json.j2 [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "Marathon": {
-    "Endpoint": "{%- for host in groups['marathon_servers'] -%}http://{% if nginx_auth_enable is defined and nginx_auth_enable|bool %}{{ marathon_bamboo_user }}:{{ marathon_bamboo_pass }}@{{ host }}:{{ nginx_http_port }}/marathon/{% else %}{% if marathon_auth_enable is defined and marathon_auth_enable|bool %}{{ marathon_bamboo_user }}:{{ marathon_bamboo_pass }}@{% endif %}{{ host }}:{{ marathon_http_port }}{% endif %}{% if not loop.last %},{% endif %}{%- endfor %}"
+    "Endpoint": "{% if bamboo_marathon_proxy is defined %}{{ bamboo_marathon_proxy_scheme }}://{{ marathon_bamboo_user }}:{{ marathon_bamboo_pass }}@{{ bamboo_marathon_proxy }}{% else %}{%- for host in groups['marathon_servers'] -%}http://{% if nginx_auth_enable is defined and nginx_auth_enable|bool %}{{ marathon_bamboo_user }}:{{ marathon_bamboo_pass }}@{{ host }}:{{ nginx_http_port }}/marathon/{% else %}{% if marathon_auth_enable is defined and marathon_auth_enable|bool %}{{ marathon_bamboo_user }}:{{ marathon_bamboo_pass }}@{% endif %}{{ host }}:{{ marathon_http_port }}{% endif %}{% if not loop.last %},{% endif %}{%- endfor %}{% endif %}"
```
Comment (ankanm) [inline ansible/roles/bamboo/templates/production.json.j2]: Replace `{{ bamboo_marathon_proxy_scheme }}` with `{{ bamboo_marathon_proxy_scheme | default('http') }}

File: ansible/roles/marathon-server/templates/marathon.conf.j2 [hunk 3]
```diff
@@ -27,6 +27,6 @@ exec /usr/bin/marathon {% if (ssl_keystore_pass is defined) and not(ssl_keystore
 
 post-start script
   sleep 60
-  curl -X POST -H "Content-Type: application/x-www-form-urlencoded" -u {{ marathon_admin_user }}:{{ marathon_admin_password }} --data "logger=mesosphere.marathon.example&level=OFF" http://{{ marathon_webui_host }}/logging
+  curl -X POST -H "Content-Type: application/x-www-form-urlencoded" -u {{ marathon_admin_user }}:{{ marathon_admin_password }} --data "logger=mesosphere.marathon.example&level=OFF" {% if bamboo_marathon_proxy_scheme is defined %}{{ bamboo_marathon_proxy_scheme }}{% else %}http{% endif %}://{{ marathon_webui_host }}/logging
```
Comment (ankanm) [inline ansible/roles/marathon-server/templates/marathon.conf.j2]: Same as before - use default.

File: ansible/roles/marathon-server/templates/marathon.conf.j2 [hunk 4]
```diff
@@ -27,6 +27,6 @@ exec /usr/bin/marathon {% if (ssl_keystore_pass is defined) and not(ssl_keystore
 
 post-start script
   sleep 60
-  curl -X POST -H "Content-Type: application/x-www-form-urlencoded" -u {{ marathon_admin_user }}:{{ marathon_admin_password }} --data "logger=mesosphere.marathon.example&level=OFF" http://{{ marathon_webui_host }}/logging
+  curl -X POST -H "Content-Type: application/x-www-form-urlencoded" -u {{ marathon_admin_user }}:{{ marathon_admin_password }} --data "logger=mesosphere.marathon.example&level=OFF" {{ bamboo_marathon_proxy_scheme|default('http') }}://{{ marathon_webui_host }}/logging
```
Comment (ankanm) [inline ansible/roles/marathon-server/templates/marathon.conf.j2]: shouldn't this be on every host (not just the leader)?

Comment (amitbose327) [inline ansible/roles/marathon-server/templates/marathon.conf.j2]: The request by any non-leader gets forwarded to the leader, and changes to the leaders reflects for all the masters.

**Comment by ankanm** [comment]
LGTM

**Pull request merged**